### ISSUE TITLE
Map Store applies aggregation in parallel

### ISSUE THREAD

**Issue by p013570** [gchq/Gaffer#811]
When elements are added it does StreamSupport.stream(elements.spliterator(), true) so it adds them in parallel. This means the aggregation is done in parallel and we could end up with the same count property being incorrectly updated at the same time in different threads.

For the time being this should be changed to not do it in parallel. In the future we could write an implementation of spliterator that splits on groups

**Comment by gaffer01** [comment]
Merged into develop.
